src/v/cluster/health_monitor_backend.h

```diff
@@ -114,8 +114,8 @@ class health_monitor_backend {
     };
```

Comment:
So most of the methods like `get_cluster_health`, `collect_current_node_health` etc still return a copy. Is that expected? Aren't those the most used methods to access the health report?

Reply:
Discussed on slack:

`get_cluster_health` returns the report struct which internall has the ptrs to the invidivual node reports.

`get_current_node_health` still returns by value but those are not used on in the metadata update path.